---
title: "OTX2 superenhancer scramble - JK data reanalysis"
author: "Pierre Murat"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    toc: yes
    toc_float: yes
    toc_collapsed: yes
    toc_depth: 3
    number_sections: yes
    theme: lumen
  pdf_document:
    toc: yes
    toc_depth: '3'
---

```{r, eval = F}
setwd("/lustre/scratch126/gengen/projects/escramble")
```

This notebook reports the (re)analysis and processing of data generated by Jonas Koeppel to generate figures reported in the OTX2 manuscript.

# Read coverage

Upload local data
scp -r /Users/pm23/Desktop/Projects/EnhancerScramble/Data/JK_data/bam_2 farm:/lustre/scratch126/gengen/projects/escramble/Data/JK_data

Two experiments are processed:
- loxPsym 6 clonal data
- loxPsym 3 (LCR - locus control region) clonal data

```{r, eval = F}

library(dplyr)
library(ggplot2)
library(forcats)

setwd("/lustre/scratch126/gengen/projects/escramble")

# Select read names

########################################
# loxP6 (clonal)

loxP6.read.names.df <- read.table("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/read_names/read_names_loxP6_clonal.tsv", header = T, sep = "\t")
unique(loxP6.read.names.df$name) # 19 architectures
# Mutate names for file manipulation
loxP6.read.names.df <- loxP6.read.names.df %>% mutate(file.name = gsub("∆", "delta", name))

# Compute the number of reads associated with each architecture
loxP6.read.count.df <- loxP6.read.names.df %>% group_by(name) %>% summarise(read.count = n())

# Prepare read name txt files

for (i in 1:length(unique(loxP6.read.count.df$name))) {
  genotype.i <- unique(loxP6.read.count.df$name)[i]
  print(genotype.i)
  read.names.i <- (loxP6.read.names.df %>% filter(name == genotype.i))$fiber
  name.i <- unique((loxP6.read.names.df %>% filter(name == genotype.i))$file.name)
  path.i <- paste0("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/", name.i, ".reads.txt")
  write.table(read.names.i, path.i, sep="\t", col.names = F, row.names = F, quote = F)
}

# WT architectures reads are defined by the group of reads that do not support a structural variant
# Save all reads supporting a structural variant
SV.read.names <- unique(loxP6.read.names.df$fiber)
path <- paste0("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/", "all.SV", ".reads.txt")
write.table(SV.read.names, path, sep="\t", col.names = F, row.names = F, quote = F)

########################################
# LCR

LCR.read.names.df <- read.table("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/read_names/read_names_LCR.tsv", header = T, sep = "\t")
unique(LCR.read.names.df$name)
# Mutate names for file manipulation
LCR.read.names.df <- LCR.read.names.df %>% mutate(file.name.temp = gsub("∆", "delta", name)) %>% mutate(file.name = gsub("\\+", "_", file.name.temp)) %>% select(-file.name.temp)

# Select genotypes supported by more than 10 reads
LCR.read.count.df <- LCR.read.names.df %>% group_by(name) %>% summarise(read.count = n()) %>% filter(read.count >= 10) # 9 genotypes

# Prepare read name txt files

for (i in 1:length(unique(LCR.read.count.df$name))) {
  genotype.i <- unique(LCR.read.count.df$name)[i]
  print(genotype.i)
  read.names.i <- (LCR.read.names.df %>% filter(name == genotype.i))$fiber
  name.i <- unique((LCR.read.names.df %>% filter(name == genotype.i))$file.name)
  path.i <- paste0("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen/", name.i, ".reads.txt")
  write.table(read.names.i, path.i, sep="\t", col.names = F, row.names = F, quote = F)
}
```

Merge bam files

```{bash, eval = F}
# FARM

# Combine bam file from all sorting gates

cd /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2

# loxP6

samtools merge -@ 12 \
-o OTX2_loxP6_all_reads.bam \
OTX2_loxP6_bright_R1_OTX2.bam OTX2_loxP6_bright_R2_OTX2.bam OTX2_loxP6_dark_R1_OTX2.bam OTX2_loxP6_dark_R2_OTX2.bam \
OTX2_loxP6_dim_R1_OTX2.bam OTX2_loxP6_dim_R2_OTX2.bam OTX2_loxP6_medium_R1_OTX2.bam OTX2_loxP6_medium_R2_OTX2.bam
samtools index -@ 12 OTX2_loxP6_all_reads.bam

# LCR

samtools merge -@ 12 \
-o OTX2_LCR_all_reads.bam \
OTX2_LCR_bright_R1_OTX2.bam OTX2_LCR_bright_R2_OTX2.bam OTX2_LCR_dark_R1_OTX2.bam OTX2_LCR_dark_R2_OTX2.bam \
OTX2_LCR_dim_R1_OTX2.bam OTX2_LCR_dim_R2_OTX2.bam OTX2_LCR_medium_R1_OTX2.bam OTX2_LCR_medium_R2_OTX2.bam
samtools index -@ 12 OTX2_LCR_all_reads.bam
```

Compute coverage for deletion variants

```{bash, eval = F}
# FARM

##################################
# loxP6

cd /lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen

samtools view /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam | grep -v -F -f all.SV.reads.txt | cut -f 1 > WT.reads.txt

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -h /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam | grep -v -F -f all.SV.reads.txt | samtools view -b -o WT.bam
samtools index -@ 12 WT.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b WT.bam -o WT.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR12345.reads.txt -o deltaR12345.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR12345.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR12345.bam -o deltaR12345.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR1234.reads.txt -o deltaR1234.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR1234.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR1234.bam -o deltaR1234.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR123.reads.txt -o deltaR123.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR123.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR123.bam -o deltaR123.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR12.reads.txt -o deltaR12.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR12.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR12.bam -o deltaR12.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR1.reads.txt -o deltaR1.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR1.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR1.bam -o deltaR1.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR34.reads.txt -o deltaR34.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR34.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR34.bam -o deltaR34.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR3.reads.txt -o deltaR3.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR3.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR3.bam -o deltaR3.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR45.reads.txt -o deltaR45.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR45.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR45.bam -o deltaR45.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR5.reads.txt -o deltaR5.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR5.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR5.bam -o deltaR5.bw'

#####
###

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N iR1234.reads.txt -o iR1234.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 iR1234.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b iR1234.bam -o iR1234.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b iR1234.bam -o iR1234.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N iR123.reads.txt -o iR123.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 iR123.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b iR123.bam -o iR123.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b iR123.bam -o iR123.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N iR12.reads.txt -o iR12.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 iR12.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b iR12.bam -o iR12.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b iR12.bam -o iR12.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N iR34.reads.txt -o iR34.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 iR34.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b iR34.bam -o iR34.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b iR34.bam -o iR34.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N iR3.reads.txt -o iR3.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 iR3.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b iR3.bam -o iR3.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b iR3.bam -o iR3.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N iR45.reads.txt -o iR45.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 iR45.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b iR45.bam -o iR45.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b iR45.bam -o iR45.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR12+iR34.reads.txt -o deltaR12_iR34.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR12_iR34.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b deltaR12_iR34.bam -o deltaR12_iR34.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b deltaR12_iR34.bam -o deltaR12_iR34.rev.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR12+iR3+deltaR4.reads.txt -o deltaR12_iR3_deltaR4.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_loxP6_all_reads.bam
samtools index -@ 12 deltaR12_iR3_deltaR4.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand reverse -b deltaR12_iR3_deltaR4.bam -o deltaR12_iR3_deltaR4.fwd.bw
bamCoverage -p 12 -bs 100 --normalizeUsing None --filterRNAstrand forward -b deltaR12_iR3_deltaR4.bam -o deltaR12_iR3_deltaR4.rev.bw'

scp -r farm:/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/ /Users/pm23/Desktop/Projects/EnhancerScramble/Data/JK_data/BW

##################################
# LCR

module load HGI/softpack/groups/escramble/eSCRAMBLE/7
cd /lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR45.reads.txt -o deltaR45.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_LCR_all_reads.bam
samtools index -@ 12 deltaR45.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR45.bam -o deltaR45.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR4.reads.txt -o deltaR4.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_LCR_all_reads.bam
samtools index -@ 12 deltaR4.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR4.bam -o deltaR4.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N deltaR5.reads.txt -o deltaR5.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_LCR_all_reads.bam
samtools index -@ 12 deltaR5.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b deltaR5.bam -o deltaR5.bw'

bsub -q normal -n 12 -R'select[mem>32000] rusage[mem=32000] span[hosts=1]' -M32000 -o coverage.log -J coverage \
'samtools view -@ 12 -m 2G -bS -N wildtype.reads.txt -o wildtype.bam /lustre/scratch126/gengen/projects/escramble/Data/JK_data/bam_2/OTX2_LCR_all_reads.bam
samtools index -@ 12 wildtype.bam
bamCoverage -p 12 -bs 100 --normalizeUsing None -b wildtype.bam -o wildtype.bw'

scp -r farm:/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen/ /Users/pm23/Desktop/Projects/EnhancerScramble/Data/JK_data/BW
```

Compute for variants containing insertions are more complex to analyse and are reported in a supporting code:
/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis_loxP6_variants.R
/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis_LCR_variants.R

Prepare files for plotting

```{r, eval = F}

##################################
# loxP6

# Define region to plot
chr <- "chr14"
from <- 56837000
to <- 56912000

# Prepare empty table for filling missing values
loxP6.coord.df <- tibble(start = seq(from+1, to+1, 100))

WT.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/WT.bw")
WT.df <- as.data.frame(WT.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "WT")

DEL_R1.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR1.bw")
DEL_R1.df <- as.data.frame(DEL_R1.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R1")

DEL_R12.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR12.bw")
DEL_R12.df <- as.data.frame(DEL_R12.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R12")

DEL_R123.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR123.bw")
DEL_R123.df <- as.data.frame(DEL_R123.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R123")

DEL_R1234.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR1234.bw")
DEL_R1234.df <- as.data.frame(DEL_R1234.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R1234")

DEL_R12345.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR12345.bw")
DEL_R12345.df <- as.data.frame(DEL_R12345.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R12345")

DEL_R3.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR3.bw")
DEL_R3.df <- as.data.frame(DEL_R3.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R3")

DEL_R34.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR34.bw")
DEL_R34.df <- as.data.frame(DEL_R34.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R34")

DEL_R45.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR45.bw")
DEL_R45.df <- as.data.frame(DEL_R45.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R45")

DEL_R5.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/loxP6_clonal_screen/deltaR5.bw")
DEL_R5.df <- as.data.frame(DEL_R5.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP6.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "DEL_R5")

loxP6.deletion.coverage.df <- rbind(WT.df, DEL_R1.df, DEL_R12.df, DEL_R123.df, DEL_R1234.df, DEL_R12345.df, DEL_R3.df, DEL_R34.df, DEL_R45.df, DEL_R5.df)
saveRDS(loxP6.deletion.coverage.df, "/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/loxP6.deletion.coverage.df.rds")

##################################
# loxP3 LCR cas9 sequencing

# Define region to plot
chr <- "chr14"
from <- 56884000
to <- 56907900

# Prepare empty table for filling missing values
loxP3.coord.df <- tibble(start = seq(from+1, to+1, 100))

# Deletion

WT.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen/wildtype.bw")
WT.df <- as.data.frame(WT.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP3.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "WT")

deltaR4.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen/deltaR4.bw")
deltaR4.df <- as.data.frame(deltaR4.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP3.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "deltaR4")

deltaR5.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen/deltaR5.bw")
deltaR5.df <- as.data.frame(deltaR5.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP3.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "deltaR5")

deltaR45.bw <- import("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/coverage/LCR_screen/deltaR45.bw")
deltaR45.df <- as.data.frame(deltaR45.bw) %>% filter(seqnames == chr & start >= from & end <= to) %>% right_join(loxP3.coord.df) %>% arrange(start) %>% fill(score) %>% mutate(variant = "deltaR45")

loxP3.deletion.cas9.coverage.df <- rbind(WT.df, deltaR4.df, deltaR5.df, deltaR45.df)
saveRDS(loxP3.deletion.cas9.coverage.df, "/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/loxP3.deletion.cas9.coverage.df.rds")
```

Plot

```{r}

library(dplyr)
library(tidyr)
library(ggplot2)
library(forcats)
library(wesanderson)
library(ggpubr)

setwd("/lustre/scratch126/gengen/projects/escramble/Notebook/")

# Define region to plot
chr <- "chr14"
from <- 56837000
to <- 56912000

# Load data

loxP6.deletion.coverage.df <- readRDS("/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/loxP6.deletion.coverage.df.rds")

loxP6.inversion.cas9.coverage.df <- readRDS("/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/loxP6.clonal.inversion.cas9.coverage.df.rds")

loxP3.deletion.cas9.coverage.df <- readRDS("/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/loxP3.deletion.cas9.coverage.df.rds")

loxP3.inversion.cas9.coverage.df <- readRDS("/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/loxP3.inversion.cas9.coverage.df.rds")

# Plot

loxP6.deletion.palette <- wes_palette("Zissou1", length(unique(loxP6.deletion.coverage.df$variant)), type = "continuous")
loxP6.deletion.coverage.plot <- loxP6.deletion.coverage.df %>% ggplot(aes(x = start/1e6, y = score, group=1, fill = variant)) +
  geom_area() +
  geom_line(col = "black", linewidth = 0.25) +
  ylab("Read count") + xlab("Coordinate [Mb]") +
  scale_fill_manual(values=c(loxP6.deletion.palette)) +
  xlim(from/1e6, to/1e6) +
    geom_vline(xintercept=56840404/1e6, linetype="dashed") +
    geom_vline(xintercept=56851954/1e6, linetype="dashed") +
    geom_vline(xintercept=56864719/1e6, linetype="dashed") +
    geom_vline(xintercept=56884635/1e6, linetype="dashed") +
    geom_vline(xintercept=56898377/1e6, linetype="dashed") +
    geom_vline(xintercept=56907609/1e6, linetype="dashed") +
  facet_wrap(~ variant, scales = "free", ncol = 3) +
  theme_bw() + theme(aspect.ratio=0.25, panel.grid.minor = element_blank(), strip.background = element_blank(), legend.position="none")
loxP6.deletion.coverage.plot

pdf("./Rplots/OTX2/loxP6.deletion.coverage.pdf", width=10, height=10, useDingbats=FALSE)
loxP6.deletion.coverage.plot
dev.off()

loxP6.inversion.cas9.palette <- wes_palette("Zissou1", length(unique(loxP6.inversion.cas9.coverage.df$sample)), type = "continuous")
loxP6.inversion.cas9.coverage.plot <- loxP6.inversion.cas9.coverage.df %>% ggplot(aes(x = start/1e6, y = score, group=1, fill = sample)) +
  geom_area() +
  geom_line(col = "black", linewidth = 0.25) +
  ylab("Read count") + xlab("Coordinate [Mb]") +
  scale_fill_manual(values=c(loxP6.inversion.cas9.palette)) +
  xlim(56837000/1e6, 56912000/1e6) +
  geom_vline(xintercept=56840404/1e6, linetype="dashed") +
  geom_vline(xintercept=56851954/1e6, linetype="dashed") +
  geom_vline(xintercept=56864719/1e6, linetype="dashed") +
  geom_vline(xintercept=56884635/1e6, linetype="dashed") +
  geom_vline(xintercept=56898377/1e6, linetype="dashed") +
  geom_vline(xintercept=56907609/1e6, linetype="dashed") +
  facet_wrap(~ sample, scales = "free", ncol = 3) +
  theme_bw() + theme(aspect.ratio=0.25, panel.grid.minor = element_blank(), strip.background = element_blank(), legend.position="none")
loxP6.inversion.cas9.coverage.plot

pdf("./Rplots/OTX2/loxP6.inversion.coverage.pdf", width=10, height=10, useDingbats=FALSE)
loxP6.inversion.cas9.coverage.plot
dev.off()

loxP3.deletion.cas9.palette <- wes_palette("Zissou1", length(unique(loxP3.deletion.cas9.coverage.df$variant)), type = "continuous")
loxP3.deletion.cas9.coverage.plot <- loxP3.deletion.cas9.coverage.df %>% ggplot(aes(x = start/1e6, y = score, group=1, fill = variant)) +
  geom_area() +
  geom_line(col = "black", linewidth = 0.25) +
  ylab("Read count") + xlab("Coordinate [Mb]") +
  scale_fill_manual(values=c(loxP3.deletion.cas9.palette)) +
  xlim(56884500/1e6, 56907900/1e6) +
    geom_vline(xintercept=56884634/1e6, linetype="dashed") +
    geom_vline(xintercept=56898376/1e6, linetype="dashed") +
    geom_vline(xintercept=56907608/1e6, linetype="dashed") +
  facet_wrap(~ variant, scales = "free", ncol = 3) +
  theme_bw() + theme(aspect.ratio=0.25, panel.grid.minor = element_blank(), strip.background = element_blank(), legend.position="none")
loxP3.deletion.cas9.coverage.plot

pdf("./Rplots/OTX2/loxP3.deletion.cas9.coverage.pdf", width=10, height=10, useDingbats=FALSE)
loxP3.deletion.cas9.coverage.plot
dev.off()

loxP3.inversion.cas9.palette <- wes_palette("Zissou1", length(unique(loxP3.inversion.cas9.coverage.df$sample)), type = "continuous")
loxP3.inversion.cas9.coverage.plot <- loxP3.inversion.cas9.coverage.df %>% ggplot(aes(x = start/1e6, y = score, group=1, fill = sample)) +
  geom_area() +
  geom_line(col = "black", linewidth = 0.25) +
  ylab("Read count") + xlab("Coordinate [Mb]") +
  scale_fill_manual(values=c(loxP3.inversion.cas9.palette)) +
  xlim(56884500/1e6, 56907900/1e6) +
  geom_vline(xintercept=56884634/1e6, linetype="dashed") +
  geom_vline(xintercept=56898376/1e6, linetype="dashed") +
  geom_vline(xintercept=56907608/1e6, linetype="dashed") +
  facet_wrap(~ sample, scales = "free", ncol = 3) +
  theme_bw() + theme(aspect.ratio=0.25, panel.grid.minor = element_blank(), strip.background = element_blank(), legend.position="none")
loxP3.inversion.cas9.coverage.plot

#### WARNING
#### iR45 and iR5 variants are not correct

pdf("./Rplots/OTX2/loxP3.inversion.cas9.coverage.pdf", width=10, height=10, useDingbats=FALSE)
loxP3.inversion.cas9.coverage.plot
dev.off()

#scp -r farm:/lustre/scratch126/gengen/projects/escramble/Notebook/Rplots/OTX2 /Users/pm23/Desktop/Projects/EnhancerScramble/Rplots
```

# Gene expression scores

Compute gene expression scores from JK data

```{r, eval=F}

library(dplyr)
library(tidyr)
library(wesanderson)

# R

#####################################
# Load df summarising all experiments performed

SV.summary.df <- read.table("/lustre/scratch126/gengen/projects/escramble/Data/JK_data/summary_table/OTX2_scramble_outcomes.tsv", sep = "\t", header = T)
# 38 variants
unique(SV.summary.df$experiment)
# 3_loxPsym_clone, 6_loxPsym_transfection, 6_loxPsym_clone 

#####################################
# loxPsym 6 clonal experiement

SV.6.loxPsym.clone.df <- SV.summary.df %>% filter(experiment == "6_loxPsym_clone")
# 20 SVs

# wrap data - Filling missing observations for plotting and score computation
genotype <- unique(SV.6.loxPsym.clone.df$variant_name)
wrap.df <-   tibble(variant_name = rep(genotype, 8)) %>% arrange(variant_name) %>% 
  mutate(gate = rep(c("dark", "dim", "medium", "bright"), 2*length(unique(SV.6.loxPsym.clone.df$variant_name)))) %>% 
  mutate(replicate = rep(c(rep(1,4), rep(2,4)), length(unique(SV.6.loxPsym.clone.df$variant_name))))
SV.6.loxPsym.clone.wrap.df <- SV.6.loxPsym.clone.df %>% full_join(wrap.df, by = c("variant_name", "gate", "replicate")) %>%
  mutate(perc_reads = if_else(is.na(perc_reads), 0, perc_reads))
saveRDS(SV.6.loxPsym.clone.wrap.df, "/lustre/scratch126/gengen/projects/escramble/Notebook/003_OTX2_superenhancer_scramble_reanalysis/rds/SV.6.loxPsym.clone.wrap.df.rds")

SV.6.loxPsym.clone.clean.df <- SV.6.loxPsym.clone.wrap.df %>%
  dplyr::select(name = variant_name, genotype = variant_definition, replicate, gate, read_count = supp_reads, read_percentage = perc_reads) %>%
  mutate(read_count = replace_na(read_count, 0)) %>% arrange(name)
write.table(SV.6.loxPsym.clone..df, "./Summary/OTX2/loxPsym.6.clonal.cas9.summary.tsv", sep="\t", col.names = T, row.names = F, quote = F)

# Plot read distributions

SV.6.loxPsym.palette <- wes_palette("Zissou1", length(unique(SV.6.loxPsym.clone.wrap.df$variant_name)), type = "continuous")
SV.6.loxPsym.gate.plot <- SV.6.loxPsym.clone.wrap.df %>% 
  mutate(SAMPLE = fct_relevel(gate, "dark", "dim", "medium", "bright")) %>%
  ggplot(aes(x=SAMPLE, y=perc_reads, group=1, col = variant_name)) +
  geom_point(aes(x = SAMPLE, y = perc_reads), size = 1) +
  geom_point(stat='summary', fun.y=sum, size = 2) +
  geom_line(stat='summary', fun.y=sum) +
  ylab("Percent of events in sorting gate") +
  scale_colour_manual(values=c(SV.6.loxPsym.palette)) +
  facet_wrap(~ variant_name, scales = "free", ncol = 5) +
  theme_bw() + theme(aspect.ratio=1, strip.background = element_blank(), panel.grid.minor = element_blank(), axis.title.x=element_blank(), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position="none")
SV.6.loxPsym.gate.plot

pdf("./Rplots/OTX2/SV.6.loxPsym.gate.plot.pdf", width=10, height=10, useDingbats=FALSE)
SV.6.loxPsym.gate.plot
dev.off()

# Compute gene expression score

# We first exclude variant with less than observations in 2 gates and with a minimum of 25 total reads

SV.6.loxPsym.summary.df <- SV.6.loxPsym.clone.wrap.df %>% group_by(variant_name, replicate) %>% summarise(na_count = sum(is.na(supp_reads))) %>% filter(na_count <= 2)

SV.6.loxPsym.summary.df.2 <- SV.6.loxPsym.clone.wrap.df %>% group_by(variant_name, replicate) %>% summarise(count = sum(supp_reads, na.rm = T)) %>% filter(count >= 25)

SV.6.loxPsym.clone.SV.score.df <- SV.6.loxPsym.clone.wrap.df %>% 
  filter(variant_name %in% SV.6.loxPsym.summary.df$variant_name) %>%
  filter(variant_name %in% SV.6.loxPsym.summary.df.2$variant_name) %>% 
  group_by(variant_name, replicate) %>%
  mutate(scale = perc_reads/sum(perc_reads)) %>% 
  dplyr::select(variant_name, gate, replicate, scale) %>% ungroup() %>% 
  pivot_wider(names_from = gate, values_from = scale) %>%
  mutate(across(where(is.numeric), ~replace_na(., 0))) %>% 
  mutate(score = (-1*dark)+(-0.5*dim)+(0.5*medium)+(1*bright))

SV.6.loxPsym.clone.SV.score.sd.df <- SV.6.loxPsym.clone.SV.score.df %>% group_by(variant_name) %>% summarise(mean=mean(score, na.rm = T), diff=max(score, na.rm = T)-mean(score, na.rm = T))

# Plot

SV.6.loxPsym.clone.SV.score.tilt.plot <- SV.6.loxPsym.clone.SV.score.df %>% 
  ggplot() + 
  geom_bar(aes(x=reorder(variant_name, score), y=score), position = "dodge", stat = "summary", fun.y = "mean", col = "black", size = 0.25, width = 0) +
  geom_errorbar(data=SV.6.loxPsym.clone.SV.score.sd.df, aes(x=variant_name, ymin=mean-diff, ymax=mean+diff), linewidth = 0.5) +
  geom_point(aes(x=reorder(variant_name, -score), y=score), size = 1.5) +
  xlab("Gene expression score") + ylim(-1,1) +
  geom_hline(yintercept=0, linetype="solid", linewidth = 0.25) +
  theme_bw() + theme(aspect.ratio=3.236, panel.grid.major = element_blank(), axis.title.y=element_blank()) + coord_flip()
SV.6.loxPsym.clone.SV.score.tilt.plot

#####################################
# loxPsym 3 experiement

SV.3.loxPsym.df <- SV.summary.df %>% filter(experiment == "3_loxPsym_clone")
# 9 SVs

# wrap data - Filling missing observations for plotting and score computation
genotype <- unique(SV.3.loxPsym.df$variant_name)
wrap.df <-   tibble(variant_name = rep(genotype, 8)) %>% arrange(variant_name) %>% 
  mutate(gate = rep(c("dark", "dim", "medium", "bright"), 2*length(unique(SV.3.loxPsym.df$variant_name)))) %>% 
  mutate(replicate = rep(c(rep(1,4), rep(2,4)), length(unique(SV.3.loxPsym.df$variant_name))))
SV.3.loxPsym.wrap.df <- SV.3.loxPsym.df %>% full_join(wrap.df, by = c("variant_name", "gate", "replicate")) %>%
  mutate(perc_reads = if_else(is.na(perc_reads), 0, perc_reads))

SV.3.loxPsym.clone.clean.df <- SV.3.loxPsym.wrap.df %>%
  dplyr::select(name = variant_name, genotype = variant_definition, replicate, gate, read_count = supp_reads, read_percentage = perc_reads) %>%
  mutate(read_count = replace_na(read_count, 0)) %>% arrange(name)
write.table(SV.3.loxPsym.clone.clean.df, "./Summary/OTX2/loxPsym.3.clonal.cas9.summary.tsv", sep="\t", col.names = T, row.names = F, quote = F)

# Plot read distributions

SV.3.loxPsym.palette <- wes_palette("Zissou1", length(unique(SV.3.loxPsym.wrap.df$variant_name)), type = "continuous")
SV.3.loxPsym.gate.plot <- SV.3.loxPsym.wrap.df %>%
  mutate(SAMPLE = fct_relevel(gate, "dark", "dim", "medium", "bright")) %>%
  ggplot(aes(x=SAMPLE, y=perc_reads, group=1, col = variant_name)) +
  geom_point(aes(x = SAMPLE, y = perc_reads), size = 1) +
  geom_point(stat='summary', fun.y=sum, size = 2) +
  geom_line(stat='summary', fun.y=sum) +
  ylab("Percent of events in sorting gate") +
  scale_colour_manual(values=c(SV.3.loxPsym.palette)) +
  facet_wrap(~ variant_name, scales = "free", ncol = 5) +
  theme_bw() + theme(aspect.ratio=1, strip.background = element_blank(), panel.grid.minor = element_blank(), axis.title.x=element_blank(), axis.text.x = element_text(angle = 45, vjust = 1, hjust=1), legend.position="none")
SV.3.loxPsym.gate.plot

pdf("./Rplots/OTX2/SV.3.loxPsym.gate.plot.pdf", width=10, height=10, useDingbats=FALSE)
SV.3.loxPsym.gate.plot
dev.off()

# Compute gene expression score

# We first exclude variant with less than observations in 2 gates and with a minimum of 25 total reads

SV.3.loxPsym.summary.df <- SV.3.loxPsym.wrap.df %>% group_by(variant_name, replicate) %>% summarise(na_count = sum(is.na(supp_reads))) %>% filter(na_count <= 2)

SV.3.loxPsym.summary.df.2 <- SV.3.loxPsym.wrap.df %>% group_by(variant_name) %>% summarise(count = sum(supp_reads, na.rm = T)) %>% filter(count >= 25)

SV.3.loxPsym.cas9.SV.score.df <- SV.3.loxPsym.wrap.df %>%
  # filter(variant_name %in% SV.3.loxPsym.summary.df$variant_name) %>%
  # filter(variant_name %in% SV.3.loxPsym.summary.df.2$variant_name) %>% 
  group_by(variant_name, replicate) %>%
  mutate(scale = perc_reads/sum(perc_reads)) %>% 
  dplyr::select(variant_name, gate, replicate, scale) %>% ungroup() %>% 
  pivot_wider(names_from = gate, values_from = scale) %>%
  mutate(across(where(is.numeric), ~replace_na(., 0))) %>% 
  mutate(score = (-1*dark)+(-0.5*dim)+(0.5*medium)+(1*bright))

SV.3.loxPsym.cas9.SV.score.sd.df <- SV.3.loxPsym.cas9.SV.score.df %>% group_by(variant_name) %>% summarise(mean=mean(score, na.rm = T), diff=max(score, na.rm = T)-mean(score, na.rm = T))

# Plot

SV.3.loxPsym.cas9.SV.score.tilt.plot <- SV.3.loxPsym.cas9.SV.score.df %>%
  ggplot() + 
  geom_bar(aes(x=reorder(variant_name, score), y=score), position = "dodge", stat = "summary", fun.y = "mean", col = "black", size = 0.25, width = 0) +
  geom_errorbar(data=SV.3.loxPsym.cas9.SV.score.sd.df, aes(x=variant_name, ymin=mean-diff, ymax=mean+diff), linewidth = 0.5) +
  geom_point(aes(x=reorder(variant_name, -score), y=score), size = 1.5) +
  xlab("Gene expression score") + ylim(-1,1) +
  geom_hline(yintercept=0, linetype="solid", linewidth = 0.25) +
  theme_bw() + theme(aspect.ratio=3.236, panel.grid.major = element_blank(), axis.title.y=element_blank()) + coord_flip()
SV.3.loxPsym.cas9.SV.score.tilt.plot

ggarrange(SV.6.loxPsym.clone.SV.score.tilt.plot, SV.3.loxPsym.cas9.SV.score.tilt.plot, ncol = 2)

# Scale scores to WT and full deletion levels

SV.6.SV.score.means.df <- SV.6.loxPsym.clone.SV.score.df %>% group_by(variant_name) %>% summarise(mean=mean(score, na.rm = T))
# WT = 0.0608, ∆R12345 = -0.371
SV.3.SV.score.means.df <- SV.3.loxPsym.cas9.SV.score.df %>% group_by(variant_name) %>% summarise(mean=mean(score, na.rm = T))
# WT = 0.748, ∆R45 = -0.597

SV.6.loxPsym.clone.SV.score.df <- SV.6.loxPsym.clone.SV.score.df %>% mutate(score.norm = ((score+0.371)/(0.0608+0.371))*100)
SV.3.loxPsym.cas9.SV.score.df <- SV.3.loxPsym.cas9.SV.score.df %>% mutate(score.norm = ((score+0.597)/(0.748+0.597))*100) 

saveRDS(SV.6.loxPsym.clone.SV.score.df, "/lustre/scratch126/gengen/projects/escramble/Notebook/004_OTX2_LCR_scramble/rds/SV.6.loxPsym.clone.SV.score.df.rds")
saveRDS(SV.3.loxPsym.cas9.SV.score.df, "/lustre/scratch126/gengen/projects/escramble/Notebook/004_OTX2_LCR_scramble/rds/SV.3.loxPsym.cas9.SV.score.df.rds")

# Compute means and differences

SV.6.SV.score.stats.df <- SV.6.loxPsym.clone.SV.score.df %>% group_by(variant_name, replicate) %>% summarise(mean=mean(score.norm, na.rm = T), diff=max(score.norm, na.rm = T)-mean(score.norm, na.rm = T))
SV.3.SV.score.stats.df <- SV.3.loxPsym.cas9.SV.score.df %>% group_by(variant_name, replicate) %>% summarise(mean=mean(score.norm, na.rm = T), diff=max(score.norm, na.rm = T)-mean(score.norm, na.rm = T))

# Plot

SV.6.loxPsym.clone.SV.score.plot <- SV.6.loxPsym.clone.SV.score.df %>% 
  ggplot() + 
  geom_bar(aes(x=reorder(variant_name, score.norm), y=score.norm), position = "dodge", stat = "summary", fun.y = "mean", col = "black", size = 0.25, width = 0) +
  geom_errorbar(data=SV.6.SV.score.stats.df, aes(x=variant_name, ymin=mean-diff, ymax=mean+diff), linewidth = 0.5) +
  geom_point(aes(x=reorder(variant_name, -score.norm), y=score.norm), size = 1.5) +
  xlab("Gene expression score") +
  geom_hline(yintercept=0, linetype="dashed", linewidth = 0.25) +
  geom_hline(yintercept=100, linetype="dashed", linewidth = 0.25) +
  scale_y_continuous("Relative gene expression", breaks = seq(0, 300, by = 50), limits = c(-20, 300)) +
  theme_bw() + theme(aspect.ratio=1.1, panel.grid.major = element_blank(), axis.title.y=element_blank()) + coord_flip()

SV.3.loxPsym.cas9.SV.score.plot <- SV.3.loxPsym.cas9.SV.score.df %>% 
  ggplot() + 
  geom_bar(aes(x=reorder(variant_name, score.norm), y=score.norm), position = "dodge", stat = "summary", fun.y = "mean", col = "black", size = 0.25, width = 0) +
  geom_errorbar(data=SV.3.SV.score.stats.df, aes(x=variant_name, ymin=mean-diff, ymax=mean+diff), linewidth = 0.5) +
  geom_point(aes(x=reorder(variant_name, -score.norm), y=score.norm), size = 1.5) +
  xlab("Gene expression score") + 
  geom_hline(yintercept=0, linetype="dashed", linewidth = 0.25) +
  geom_hline(yintercept=100, linetype="dashed", linewidth = 0.25) +
  scale_y_continuous("Relative gene expression", breaks = seq(0, 300, by = 50), limits = c(-20, 300)) +
  theme_bw() + theme(aspect.ratio=1.5, panel.grid.major = element_blank(), axis.title.y=element_blank()) + coord_flip()

ggarrange(SV.6.loxPsym.clone.SV.score.plot, SV.3.loxPsym.cas9.SV.score.plot, ncol = 2)

pdf("./Rplots/OTX2/SV.6.3.gene.expression.score.pdf", width=10, height=10, useDingbats=FALSE)
ggarrange(SV.6.loxPsym.clone.SV.score.plot, SV.3.loxPsym.cas9.SV.score.plot, ncol = 2)
dev.off()
```


























